src/cpp/include/openvino/genai/rag/text_rerank_pipeline.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "openvino/genai/tokenizer.hpp"
+
+namespace ov {
+namespace genai {
+
+class OPENVINO_GENAI_EXPORTS TextRerankPipeline {
+public:
+    struct OPENVINO_GENAI_EXPORTS Config {
+        /**
+         * @brief Number of documents to return sorted by score
+         */
+        size_t top_n = 3;
+
+        /**
+         * @brief Constructs text rerank pipeline configuration
+         */
+        Config() = default;
+
+        /**
+         * @brief Constructs text rerank pipeline configuration
+         *
+         * @param properties configuration options
+         *
+         * const ov::AnyMap properties{{"top_n", 3}};
+         * ov::genai::TextRerankPipeline::Config config(properties);
+         *
+         * ov::genai::TextRerankPipeline::Config config({{"top_n", 3}});
+         */
+        explicit Config(const ov::AnyMap& properties);
+    };
```

Comment:
Add `max_length`, `pad_to_max_length` and `batch_size` to the config.

Reply:
I would rather do it in a separate PR after we confirm approach is working for `TextEmbeddingPipeline`.

Reply:
Are you sure we need `max_length` in any case? Neither LlamaIndex nor LangChain has a max_length parameter.